### ISSUE TITLE
Clearing selectedDays[]

### ISSUE THREAD

**Issue by Chris0lsen** [arca-computing/MultipleDatePicker#32]
Hi,

I'm using MultipleDatePicker to make updates to events attached to certain days in an $mdDialog window. I'd like to unselect all days when the modal window is dismissed, but simply saying
```
scope.selectedDays = [];
```
won't do it. Is there any built-in way to de-select given days?

Thanks!
Chris

**Comment by mgohin** [comment]
Hi,

I think you need to call `$scope.$apply()` right after `scope.selectedDays = [];`

http://jimhoskins.com/2012/12/17/angularjs-and-apply.html

> You do need to use it if you are going to run code in a new turn. And only if that turn isn’t being created from a method in the AngularJS library

**Comment by Chris0lsen** [comment]
Hi,

I gave it a shot and got the dreaded ```$digest already in progress``` error, so I wrapped my code in ```$timeout()``` like so:

```$timeout(function() {
  scope.selectedDays = [];
  scope.highlightDays = [];
}
```
It still isn't clearing, though. 

I noticed (using the ng-inspector debugger add-on in Chrome) that the multiple-date-picker itself holds a collection of days, each with a boolean "selected" value. Is there a way to access these from the parent directive?

Thanks for your help!

**Comment by mgohin** [comment]
Because you wrapped it in the `$timeout` you should be able to use `$scope.$apply()` without the digest problem ?

To get selected days you have to use `dayClick` callback like in demo to keep track of days selected/unselected, there is no built in way to get them (because we want to keep it light and flexible for everyone).

Here an example, in HTML :

    <multiple-date-picker day-click="myCallback"/>

In your controller

    var daysSelected = [];
    $scope.myCallback= function(event, date) {
        //date.selected change AFTER this callback, so date.selected is inverted
        if(!date.selected){
            daysSelected.push(date.valueOf()); //store timestamp, you can store the whole date
        }else{
            var index = daysSelected.indexOf(date.valueOf());
            daysSelected.splice(index, 1);
        }
    }

**Comment by Chris0lsen** [comment]
I'll keep working on this today, but I don't think I'm communicating my problem properly: The user selects multiple days in order to make changes to all of them at once (this part of my application is working; I have an array that holds all the selected days). After they make the changes, I want to set date.selected to false for all days in my selectedDays array - removing the css .picker-day.picker-selected class and turning the color back to normal.

Could I have a function in my directive to do this, or would I need to extend the multipleDatePicker.js in my project to achieve this?

Thank you, again, for your help!

**Comment by mgohin** [comment]
I'm pretty sure we are on the same line but 

>  I want to set date.selected to false for all days in my selectedDays array

you want to reset the calendar ?

if so you need to do this

    $timeout(function() {
        scope.selectedDays = []; //beware that directive property is daysSelected
        scope.$apply(); //you shouldn't have the digest problem
    }

Your problem is just to apply the change in the MultipleDatePicker directive

**Comment by Chris0lsen** [comment]
Still having some difficulty here. I bound multiple-date-picker's daysSelected to my selectedDays, thinking that it would trigger the ```scope.$watch('daysSelected)``` function, but it isn't; even with ```days-selected="selectedDays"``` as an attribute in the multiple-date-picker element.

I tried the code you suggested, but the multiple-date-picker's convertedDaysSelected array still held objects after I ran the ```scope.$apply()```.

I really appreciate your patience. I think multiple-date-picker is a great tool, and my difficulty is coming from my own failure to understand nesting directives in Angular.
